examples/main.cpp

```diff
@@ -699,6 +821,7 @@ int main(int argc, char **argv)
     }
     CALIPER(CALI_MARK_END("Cycle");)
     MPI_CALL(MPI_Barrier(MPI_COMM_WORLD));
+    printMemory(allocator_names);
```

Comment:
Is printing always memory info intentional? Alternative is to guard it only for debugging

Reply:
We currently use the example code as a very exhaustive test. I think we should always print out the memory consumption of all of our allocators.